scrapy/utils/misc.py

```diff
@@ -242,7 +242,11 @@ def warn_on_generator_with_return_value(spider, callable):
     Logs a warning if a callable is a generator function and includes
     a 'return' statement with a value different than None
     """
-    if is_generator_with_return_value(callable):
+    try:
+        should_warn = is_generator_with_return_value(callable)
+    except SyntaxError:  # includes IndentationError
```

Comment:
Here it would be better to capture `except SyntaxError as e:` and find a way to include `e` in the reporting.

Reply:
I beg to differ, this function only takes care of warning if the passed function is a generator with a return value different than `None`. Other pieces of code will fail loudly in case of `SyntaxError` (and probably even fail earlier, this is likely a safety web that's not even going to be reached).